packages/flutter/lib/src/material/material.dart

```diff
@@ -77,27 +66,13 @@ const Map<MaterialType, BorderRadius?> kMaterialEdges = <MaterialType, BorderRad
   MaterialType.transparency: null,
 };
 
-/// An interface for creating [InkSplash]s and [InkHighlight]s on a [Material].
-///
-/// Typically obtained via [Material.of].
-abstract class MaterialInkController {
-  /// The color of the material.
-  Color? get color;
-
-  /// The ticker provider used by the controller.
-  ///
-  /// Ink features that are added to this controller with [addInkFeature] should
-  /// use this vsync to drive their animations.
-  TickerProvider get vsync;
-
-  /// Add an [InkFeature], such as an [InkSplash] or an [InkHighlight].
-  ///
-  /// The ink feature will paint as part of this controller.
-  void addInkFeature(InkFeature feature);
-
-  /// Notifies the controller that one of its ink features needs to repaint.
-  void markNeedsPaint();
-}
+@Deprecated(
+  'Use SplashController instead. '
+  'Both Material and SplashBox can be used for Splash effects. '
+  'This feature was deprecated after v3.24.0-0.2.pre.',
+)
+/// An interface for creating ink effects on a [Material].
+typedef MaterialInkController = SplashController;
```

Comment:
Rather than deprecating and typedef-ing this, can it implement the interface and add the `addInkFeature`? That way it would not break anyone, and we would not need to add that method as a deprecation in the new class.

Reply:
I personally am not worried about breaking anyone here, since data-driven fixes are easy and since interacting with a `MaterialInkController` directly is highly unusual.

(As far as implementing the interface, [see comment below](https://github.com/flutter/flutter/pull/149963#discussion_r1783358574))